### PR TITLE
feat: Read from new `team_settings` table

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Reads from new `team_settings` table instead of `team.settings` column throughout the application
 - Import `Team` type from planx-core in the API (and many associate type fixes)
 - Relies on https://github.com/theopensystemslab/planx-core/pull/444
 
 ## Next steps...
 - Write to new table in `team.create()` (planx-core)
 - Drop `team.settings` column

### CODE REVIEW

File: hasura.planx.uk/migrations/1720511391883_run_sql_migration/up.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    t.settings->>'homepage' as homepage,
+    t.domain as subdomain,
+    ti.has_planning_data as planning_data_enabled,
+    '@todo' as article_4s_enabled,  
+    jsonb_build_object(
+        'helpEmail', ts.help_email,
+        'helpPhone', ts.help_phone,
+        'emailReplyToId', ts.email_reply_to_id,
+        'helpOpeningHours', ts.help_opening_hours
+        ) as govnotify_personalisation,
```
Comment (DafyddLlyr) [inline hasura.planx.uk/migrations/1720511391883_run_sql_migration/up.sql:15]: Here's the change in this view, the rest stays the same.

Comment (DafyddLlyr) [inline hasura.planx.uk/migrations/1720511391883_run_sql_migration/up.sql:15]: This is the new section of this view - constructing the same shape JSONB column from the new `team_settings` column.

File: api.planx.uk/modules/pay/service/inviteToPay/sendConfirmationEmail.ts [hunk 2]
```diff
@@ -47,7 +46,12 @@ async function getDataForPayeeAndAgentEmails(
         flow {
           slug
           team {
-            notifyPersonalisation: notify_personalisation
+            notifyPersonalisation: team_settings {
```
Comment (DafyddLlyr) [inline api.planx.uk/modules/pay/service/inviteToPay/sendConfirmationEmail.ts:49]: Where queries are using a custom type (not a shared `Team` type from planx-core), I've kept the contextual variable name previously used to minimise changes in this already noisy PR.

**Review by jessicamcinchak — APPROVED** [review]
All looks good, thanks for working through these changes !

**Comment by DafyddLlyr** [comment]
> I'm getting "Not found" when navigating to any "Team settings" menu - unclear if introduced here, but is this a known issue? Eg https://3391.planx.pizza/barnet/settings/general is logging error TypeError: Cannot read properties of undefined (reading 'homepage')

Thanks for spotting, fixed in [cedb7a6](https://github.com/theopensystemslab/planx-new/pull/3391/commits/cedb7a6ae52ae46566718ad4f489a5be4fb14834). Not sure if it was a rebase in planx-new or just a plain oversight, but fixed now!

**Pull request merged**